compiler/qsc_frontend/src/lower.rs

```diff
@@ -429,29 +438,40 @@ impl With<'_> {
             adj,
             ctl,
             ctl_adj,
+            attrs: attrs.to_vec(),
+        }
+    }
+
+    fn check_invalid_attrs_on_function(&mut self, attrs: &[hir::Attr], span: Span) {
+        const INVALID_ATTRS: [hir::Attr; 2] = [hir::Attr::Measurement, hir::Attr::Reset];
+
+        for invalid_attr in &INVALID_ATTRS {
+            if attrs.contains(invalid_attr) {
+                self.lowerer.errors.push(Error::InvalidAttrOnFunction(
+                    format!("{invalid_attr:?}"),
+                    span,
+                ));
+            }
         }
     }
 
     fn lower_callable_kind(
         &mut self,
         kind: ast::CallableKind,
-        attrs: &[qsc_hir::hir::Attr],
+        attrs: &[hir::Attr],
         span: Span,
     ) -> hir::CallableKind {
-        if attrs.contains(&qsc_hir::hir::Attr::Measurement) {
-            match kind {
-                ast::CallableKind::Operation => hir::CallableKind::Measurement,
-                ast::CallableKind::Function => {
-                    self.lowerer
-                        .errors
-                        .push(Error::InvalidMeasurementAttrOnFunction(span));
-                    hir::CallableKind::Function
-                }
+        match kind {
+            ast::CallableKind::Function => {
+                self.check_invalid_attrs_on_function(attrs, span);
+                hir::CallableKind::Function
             }
```

Comment:
Should this be checked during lowering or should it be part of the `qsc_passes` checking the semantics of Measurements and Resets?

Reply:
I think it's ok for this one to be checked during lowering, though I suppose it could go either way. I don't feel strongly about it.

Reply:
I will have a better answer for this once Measurements and Resets are unified.